### ISSUE TITLE
Some questions about process_lrs3 environment

### ISSUE THREAD

**Issue by hylq66** [yerfor/GeneFace#40]
Hi, I'm having a problem with tensorflow when running python data_gen/nerf/extract_3dmm.py in process_lrs3. Is this normal?
![image](https://user-images.githubusercontent.com/113688724/222711439-3f1eeec4-47c7-4d79-96d2-77dddd8d008b.png)
0.png)

My environment is python=3.7.11, torch=1.11.0+cu113, tensorflow=1.14.0
![image](https://user-images.githubusercontent.com/113688724/222605810-c7a1af45-11df-4142-86b5-05440065cd89.png)

**Comment by yerfor** [comment]
Hi, it means your tensorflow is not compiled with CUDA, it is expected when you are not using old GPUs like RTX2080. 
Unfortunately the FaceRecon we use is based on tf1, so if you use newer GPUs, it only uses CPU when extracting 3DMM coefficient, which leads a slower data preprocessing process, but no errors.

**Comment by hylq66** [comment]
Hi, I'm having a problem with tensorflow when running python data_gen/nerf/extract_3dmm.py in process_lrs3. The following error should be unable to find the GPU, have you ever encountered this situation? Does tensorflow require a GPU version?

![1677843393082](https://user-images.githubusercontent.com/113688724/222711053-eebcba46-3b31-425c-99e8-f521bf2dd690.png)

**Comment by hylq66** [comment]
Thanks.

**Comment by yerfor** [comment]
Hi, I have tried run `python data_gen/nerf/extract_3dmm.py` in process_lrs3, and it generate coeff.npy correctly. My device is RTX3090. Below is my python env details:

>  packages in environment at /home/yezhenhui/anaconda3/envs/process_lrs3:
> Name                    Version                   Build  Channel
> _libgcc_mutex             0.1                        main    defaults
> _openmp_mutex             5.1                       1_gnu    defaults
> _tflow_select             2.3.0                       mkl    defaults
> absl-py                   0.7.1                    pypi_0    pypi
> appdirs                   1.4.4                    pypi_0    pypi
> astor                     0.8.1            py37h06a4308_0    defaults
> attrdict                  2.0.1                    pypi_0    pypi
> audioread                 3.0.0                    pypi_0    pypi
> blas                      1.0                         mkl    defaults
> brotlipy                  0.7.0           py37h27cfd23_1003    defaults
> bzip2                     1.0.8                h7b6447c_0    defaults
> c-ares                    1.18.1               h7f8727e_0    defaults
> ca-certificates           2023.01.10           h06a4308_0    defaults
> certifi                   2022.12.7        py37h06a4308_0    defaults
> cffi                      1.15.1           py37h74dc2b5_0    defaults
> charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
> cryptography              38.0.4           py37h9ce1e76_0    defaults
> cudatoolkit               11.3.1               h2bc3f7f_2    defaults
> decorator                 4.4.2                    pypi_0    pypi
> ds-ctcdecoder             0.9.0a10                 pypi_0    pypi
> easydict                  1.10                     pypi_0    pypi
> einops                    0.6.0                    pypi_0    pypi
> face-alignment            1.2.0                    pypi_0    pypi
> ffmpeg                    4.3                  hf484d3e_0    pytorch
> fftw                      3.3.9                h27cfd23_1    defaults
> filelock                  3.9.0                    pypi_0    pypi
> flit-core                 3.6.0              pyhd3eb1b0_0    defaults
> freetype                  2.12.1               h4a9f257_0    defaults
> gast                      0.2.2                    pypi_0    pypi
> giflib                    5.2.1                h5eee18b_3    defaults
> gmp                       6.2.1                h295c915_3    defaults
> gnutls                    3.6.15               he1e5248_0    defaults
> google-pasta              0.2.0              pyhd3eb1b0_0    defaults
> grpcio                    1.42.0           py37hce63b2e_0    defaults
> h5py                      3.7.0            py37h737f45e_0    defaults
> hdf5                      1.10.6               h3ffc7dd_1    defaults
> huggingface-hub           0.12.1                   pypi_0    pypi
> idna                      3.4              py37h06a4308_0    defaults
> imageio                   2.9.0                    pypi_0    pypi
> imageio-ffmpeg            0.4.8                    pypi_0    pypi
> importlib-metadata        4.11.3           py37h06a4308_0    defaults
> intel-openmp              2021.4.0          h06a4308_3561    defaults
> joblib                    1.2.0                    pypi_0    pypi
> jpeg                      9e                   h7f8727e_0    defaults
> keras-applications        1.0.8                      py_1    defaults
> keras-preprocessing       1.1.2              pyhd3eb1b0_0    defaults
> lame                      3.100                h7b6447c_0    defaults
> lcms2                     2.12                 h3be6417_0    defaults
> ld_impl_linux-64          2.38                 h1181459_1    defaults
> lerc                      3.0                  h295c915_0    defaults
> libdeflate                1.8                  h7f8727e_5    defaults
> libffi                    3.3                  he6710b0_2    defaults
> libgcc-ng                 11.2.0               h1234567_1    defaults
> libgfortran-ng            11.2.0               h00389a5_1    defaults
> libgfortran5              11.2.0               h1234567_1    defaults
> libgomp                   11.2.0               h1234567_1    defaults
> libiconv                  1.16                 h7f8727e_2    defaults
> libidn2                   2.3.2                h7f8727e_0    defaults
> libpng                    1.6.37               hbc83047_0    defaults
> libprotobuf               3.20.3               he621ea3_0    defaults
> librosa                   0.8.0                    pypi_0    pypi
> libstdcxx-ng              11.2.0               h1234567_1    defaults
> libtasn1                  4.16.0               h27cfd23_0    defaults
> libtiff                   4.5.0                h6a678d5_1    defaults
> libunistring              0.9.10               h27cfd23_0    defaults
> libuv                     1.40.0               h7b6447c_0    defaults
> libwebp                   1.2.4                h11a3e52_0    defaults
> libwebp-base              1.2.4                h5eee18b_0    defaults
> llvmlite                  0.39.1                   pypi_0    pypi
> lmdb                      1.2.1                    pypi_0    pypi
> lz4-c                     1.9.4                h6a678d5_0    defaults
> markdown                  3.4.1            py37h06a4308_0    defaults
> markupsafe                2.1.1            py37h7f8727e_0    defaults
> mkl                       2021.4.0           h06a4308_640    defaults
> mkl-service               2.4.0            py37h7f8727e_0    defaults
> mkl_fft                   1.3.1            py37hd3c417c_0    defaults
> mkl_random                1.2.2            py37h51133e4_0    defaults
> moviepy                   1.0.3                    pypi_0    pypi
> ncurses                   6.4                  h6a678d5_0    defaults
> nettle                    3.7.3                hbbd107a_1    defaults
> networkx                  2.6.3                    pypi_0    pypi
> numba                     0.56.4                   pypi_0    pypi
> numpy                     1.21.5           py37h6c91a56_3    defaults
> numpy-base                1.21.5           py37ha15fc14_3    defaults
> opencv-python             4.7.0.68                 pypi_0    pypi
> openh264                  2.1.1                h4ff587b_0    defaults
> openssl                   1.1.1t               h7f8727e_0    defaults
> packaging                 23.0                     pypi_0    pypi
> pillow                    9.3.0            py37h6a678d5_2    defaults
> pip                       22.3.1           py37h06a4308_0    defaults
> pooch                     1.6.0                    pypi_0    pypi
> praat-parselmouth         0.3.3                    pypi_0    pypi
> proglog                   0.1.10                   pypi_0    pypi
> progressbar               2.5                      pypi_0    pypi
> protobuf                  3.20.3           py37h6a678d5_0    defaults
> pycparser                 2.21               pyhd3eb1b0_0    defaults
> pyopenssl                 22.0.0             pyhd3eb1b0_0    defaults
> pysocks                   1.7.1                    py37_1    defaults
> python                    3.7.11               h12debd9_0    defaults
> pytorch                   1.11.0          py3.7_cuda11.3_cudnn8.2.0_0    pytorch
> pytorch-mutex             1.0                        cuda    pytorch
> pywavelets                1.3.0                    pypi_0    pypi
> pyxdg                     0.21                     pypi_0    pypi
> pyyaml                    6.0                      pypi_0    pypi
> readline                  8.2                  h5eee18b_0    defaults
> regex                     2022.10.31               pypi_0    pypi
> requests                  2.28.1           py37h06a4308_0    defaults
> resampy                   0.4.2                    pypi_0    pypi
> scikit-image              0.19.3                   pypi_0    pypi
> scikit-learn              1.0.2                    pypi_0    pypi
> scipy                     1.7.3            py37h6c91a56_2    defaults
> semver                    2.13.0                   pypi_0    pypi
> setuptools                65.6.3           py37h06a4308_0    defaults
> six                       1.15.0                   pypi_0    pypi
> soundfile                 0.12.1                   pypi_0    pypi
> sqlite                    3.40.1               h5082296_0    defaults
> tensorboard               1.14.0           py37hf484d3e_0    defaults
> tensorboardx              2.6                      pypi_0    pypi
> tensorflow                1.14.0          mkl_py37h45c423b_0    defaults
> tensorflow-addons         0.19.0                   pypi_0    pypi
> tensorflow-base           1.14.0          mkl_py37h7ce6ba3_0    defaults
> tensorflow-estimator      1.14.0                     py_0    defaults
> termcolor                 2.1.0            py37h06a4308_0    defaults
> threadpoolctl             3.1.0                    pypi_0    pypi
> tifffile                  2021.11.2                pypi_0    pypi
> tk                        8.6.12               h1ccaba5_0    defaults
> tokenizers                0.13.2                   pypi_0    pypi
> torchvision               0.12.0               py37_cu113    pytorch
> tqdm                      4.64.1                   pypi_0    pypi
> transformers              4.26.0                   pypi_0    pypi
> typeguard                 2.13.3                   pypi_0    pypi
> typing_extensions         4.4.0            py37h06a4308_0    defaults
> urllib3                   1.26.14          py37h06a4308_0    defaults
> vector-quantize-pytorch   0.4.10                   pypi_0    pypi
> werkzeug                  2.2.2            py37h06a4308_0    defaults
> wheel                     0.38.4           py37h06a4308_0    defaults
> wrapt                     1.14.1           py37h5eee18b_0    defaults
> xz                        5.2.10               h5eee18b_1    defaults
> zipp                      3.11.0           py37h06a4308_0    defaults
> zlib                      1.2.13               h5eee18b_0    defaults
> zstd                      1.5.2                ha4553b6_0    defaults

**Comment by hylq66** [comment]
Hi, I reconfigured the environment, and now the various packages are the same as yours, except the minor versions may be a little different. Do you encounter the following problem after installing tensorflow with conda install tensorflow==1.14.0? Can you share the solution?

![image](https://user-images.githubusercontent.com/113688724/223002335-2e6352e7-64bd-4452-89ab-dc2025a94d92.png)

![image](https://user-images.githubusercontent.com/113688724/223002226-495c3e1b-3bc7-495d-976f-2463420eb66a.png)

**Comment by yerfor** [comment]
Hi, this means you need re-compile the mesh renderer, since the tf-version you are using is different from the previous tf-version that you compiled the mesh renderer.
